rust/cardano-blockchain-types/src/point.rs

```diff
@@ -0,0 +1,595 @@
+        write!(f, "Point @ {slot}:{}", hex::encode(hash))
+    }
+}
+
+impl From<pallas::network::miniprotocols::Point> for Point {
+    fn from(point: pallas::network::miniprotocols::Point) -> Self {
+        Self(point)
+    }
+}
+
+impl From<Point> for pallas::network::miniprotocols::Point {
+    fn from(point: Point) -> pallas::network::miniprotocols::Point {
+        point.0
+    }
+}
+
+impl PartialOrd for Point {
+    /// Implements a partial ordering based on the slot number
+    /// of two `Point` instances. It only checks the slot number for ordering.
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Point {
+    /// Implements a total ordering based on the slot number
+    /// of two `Point` instances. It only checks the slot number for ordering.
+    fn cmp(&self, other: &Self) -> Ordering {
+        cmp_point(&self.0, &other.0)
+    }
+}
+
+impl PartialEq<u64> for Point {
+    /// Allows to compare a `SnapshotID` against `u64` (Just the Immutable File Number).
+    ///
+    /// Equality ONLY checks the Immutable File Number, not the path.
+    /// This is because the Filename is already the Immutable File Number.
+    fn eq(&self, other: &u64) -> bool {
+        self.0.slot_or_default() == *other
+    }
+}
+
+impl PartialOrd<u64> for Point {
+    /// Allows to compare a `Point` against a `u64` (Just the Immutable File Number).
+    ///
+    /// Equality ONLY checks the Immutable File Number, not the path.
+    /// This is because the Filename is already the Immutable File Number.
+    fn partial_cmp(&self, other: &u64) -> Option<Ordering> {
+        self.0.slot_or_default().partial_cmp(other)
+    }
+}
+
+impl PartialEq<Option<Point>> for Point {
+    /// Allows to compare a `SnapshotID` against `u64` (Just the Immutable File Number).
+    ///
+    /// Equality ONLY checks the Immutable File Number, not the path.
+    /// This is because the Filename is already the Immutable File Number.
+    fn eq(&self, other: &Option<Point>) -> bool {
+        if let Some(other) = other {
+            *self == *other
+        } else {
+            false
+        }
+    }
+}
+
+impl PartialOrd<Option<Point>> for Point {
+    /// Allows to compare a `Point` against a `u64` (Just the Immutable File Number).
+    ///
+    /// Equality ONLY checks the Immutable File Number, not the path.
+    /// This is because the Filename is already the Immutable File Number.
+    /// Any point is greater than None.
+    fn partial_cmp(&self, other: &Option<Point>) -> Option<Ordering> {
+        if let Some(other) = other {
+            self.partial_cmp(other)
+        } else {
+            Some(Ordering::Greater)
+        }
+    }
+}
+
+impl Default for Point {
+    /// Returns the default value for `Point`, which is `UNKNOWN_POINT`.
+    fn default() -> Self {
+        Self::UNKNOWN
+    }
+}
+
+/// Compare Points, because Pallas does not impl `Ord` for Point.
+pub(crate) fn cmp_point(
```

Comment:
As far as I can see, this function isn't used anywhere else, so `pub(crate)` can be removed.

Reply:
Its used by `impl Ord for Point` does not need to be `pub` anything.  Should be private inside this module.